ui/server/src/test/scala/pl/touk/nussknacker/ui/api/ProcessesResourcesSpec.scala

```diff
@@ -343,36 +344,30 @@ class ProcessesResourcesSpec extends FunSuite with ScalatestRouteTest with Match
     val secondProcessor = ProcessName("Processor2")
     val thirdProcessor = ProcessName("Processor3")
 
-    createProcess(firstProcessor, testCategoryName, false)
-    createDeployedCanceledProcess(secondProcessor, testCategoryName, false)
-    createDeployedProcess(thirdProcessor, testCategoryName, false)
+    createProcess(firstProcessor)
+    createDeployedCanceledProcess(secondProcessor)
+    createDeployedProcess(thirdProcessor)
 
-    Get(s"/processes") ~> routeWithAllPermissions ~> check {
-      status shouldEqual StatusCodes.OK
-      val data = responseAs[List[Json]]
-      data.size shouldBe 3
-
-      val process = findJsonProcess(responseAs[String], firstProcessor.value)
-      process.value.stateStatus shouldBe Some(SimpleStateStatus.NotDeployed.name)
+    withProcesses(ProcessesQuery.empty) { processes =>
+      processes.size shouldBe 3
+      val status = processes.find(_.name == firstProcessor.value).flatMap(_.stateStatus)
+      status shouldBe Some(SimpleStateStatus.NotDeployed.name)
     }
 
-    Get(s"/processes?isDeployed=true") ~> routeWithAllPermissions ~> check {
-      status shouldEqual StatusCodes.OK
-      responseAs[List[Json]].size shouldBe 1
-
-      val process = findJsonProcess(responseAs[String], thirdProcessor.value)
-      process.value.name shouldBe thirdProcessor.value
-      process.value.stateStatus shouldBe Some(SimpleStateStatus.Running.name)
+    withProcesses(ProcessesQuery.deployed()) { processes =>
+      processes.size shouldBe 1
+      val status = processes.find(_.name == thirdProcessor.value).flatMap(_.stateStatus)
+      status shouldBe Some(SimpleStateStatus.Running.name)
     }
 
-    Get(s"/processes?isDeployed=false") ~> routeWithAllPermissions ~> check {
-      status shouldEqual StatusCodes.OK
-      responseAs[List[Json]].size shouldBe 2
+    withProcesses(ProcessesQuery.notDeployed()) { processes =>
```

Comment:
I'd change name of this method, like: forScenariosReturned, currently it more sounds like we're mocking sth